frontend/src/metabase/lib/favicon.js

```diff
@@ -0,0 +1,22 @@
+import _ from "lodash";
+
+const DEFAULT_FAVICON = "app/assets/img/favicon.ico";
+
+export const flashFavicon = _.debounce((icon, timeout) => {
+  document.querySelector('link[rel="icon"]').setAttribute("href", icon);
+  if (timeout) {
+    setTimeout(() => {
+      document
+        .querySelector('link[rel="icon"]')
+        .setAttribute("href", DEFAULT_FAVICON);
+    }, timeout);
+  }
+});
+
+export const resetFavicon = _.debounce((delay = 0) => {
+  setTimeout(() => {
+    document
+      .querySelector('link[rel="icon"]')
+      .setAttribute("href", DEFAULT_FAVICON);
+  }, delay);
+});
```

Comment:
Why wrap functions with `_.debounce` without providing the debounce timeout as a second argument?

Reply:
Also, I wonder if `setTimeout` should be better done in components using these functions:

1. If we want to customize a favicon for other pages, they won't necessarily need the timeout
2. It'd be easier for a particular component to save the timeout ID and then do `clearTimeout` before unmounting to avoid memory leaks and unexpected favicon changes. Right now, if you call `flashFavicon` with e.g. 5-sec timeout and go to another page, the favicon will change somewhere else.

Reply:
> Why wrap functions with `_.debounce` without providing the debounce timeout as a second argument?

Haha, I honestly thought that maybe underscore had a default debounce timeout because of what I saw [here](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/hoc/Title.jsx#L9). But I'll go ahead and add 300ms or something. The debounce may not even be super necessary, but as it's called in a componentDidUpdate I thought it might be a good diea.

> Also, I wonder if setTimeout should be better done in components using these functions

We could do that. I was thinking that having a central location where the favicon is managed may be easier, but that may not be the case. With regards to clearing the timeout, I could always return the timeout ID from the function for the caller to then use.

Reply:
> Haha, I honestly thought that maybe underscore had a default debounce timeout because of what I saw [here](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/hoc/Title.jsx#L9). But I'll go ahead and add 300ms or something. The debounce may not even be super necessary, but as it's called in a componentDidUpdate I thought it might be a good idea.

Oh right, got it. Yeah, I think it'd be better to have the param set explicitly to avoid confusion